### DISCUSSION TITLE
Plans for GA version releases

### DISCUSSION THREAD

**Discussion by KimDoubleB** [open-telemetry/opentelemetry-java-instrumentation#12078] [Q&A] [answered]
Hello.

I am interested in Observability and recently learning to apply openTelemetry to services.
There are many spring boot application services currently in production, so I'm checking out the opentelemetry spring boot starter and applying it.

However, most of the instrumentation libraries are set to alpha version.

```
ex) When importing `opentelemetry-spring-boot-autoconfigure:2.6.0`
- io.opentelemetry.instrumentation:opentelemetry-instrumentation-annotations-support:2.6.0-alpha
- io.opentelemetry.instrumentation:opentelemetry-kafka-clients-2.6:2.6.0-alpha
- io.opentelemetry.instrumentation:opentelemetry-mongo-3.1:2.6.0-alpha
```

I was wondering about this, and saw that a maintainer already talked about it in `discussion` (https://github.com/open-telemetry/opentelemetry-java-instrumentation/discussions/11650#discussioncomment-9833393).
>  it's because the OpenTelemetry Spring Boot Starter is not (quite) GA. we are currently targeting July release to mark it stable 🤞

<br/>

**Question**

- Since July has passed, is there a new GA schedule?
  - I saw [the project for spring boot GA](https://github.com/orgs/open-telemetry/projects/71), but I was wondering if there is a set new schedule.
- Are there any plans to release GA versions of the opentelemetry instrumentation libraries? (like [spring boot support](https://spring.io/projects/spring-boot#support))

**Comment by laurit** [comment, marked as answer]
Spring boot autoconfigure module has been marked stable (as you can see the version number does not have `-alpha` suffix) but the instrumentations it uses have not been marked stable. Have a look at https://opentelemetry.io/docs/specs/otel/telemetry-stability/ to understand what is required for instrumentations to reach stable status. The first requirement is that the semantic conventions used by the instrumentation are marked stable, this far the only stable conventions are for http client and server.

**Reply by KimDoubleB** [reply]
Thanks for the answer.
I understand that the `io.opentelemetry.instrumentation` libraries are under development.

I haven't looked into the instrumentation library code in detail yet, so I have a question.
If I don't import the instrumentation library, it won't be automated to log/trace/metric certain situations like spring boot web/jdbc/kafka/... right?
